internal/utils/tls.go

```diff
@@ -0,0 +1,122 @@
+package utils
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"time"
+
+	interfaces "github.com/llm-d-incubation/inferno-autoscaler/internal/interfaces"
+	"github.com/llm-d-incubation/inferno-autoscaler/internal/logger"
+)
+
+// CreateTLSConfig creates a TLS configuration from PrometheusConfig
+func CreateTLSConfig(promConfig *interfaces.PrometheusConfig) (*tls.Config, error) {
+	if promConfig == nil || !promConfig.EnableTLS {
+		return nil, nil
+	}
+
+	config := &tls.Config{
+		InsecureSkipVerify: promConfig.InsecureSkipVerify,
+		ServerName:         promConfig.ServerName,
+		MinVersion:         tls.VersionTLS12, // Enforce minimum TLS version
+	}
+
+	// Load CA certificate if provided
+	if promConfig.CACertPath != "" {
+		caCert, err := os.ReadFile(promConfig.CACertPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read CA certificate from %s: %w", promConfig.CACertPath, err)
+		}
+
+		caCertPool := x509.NewCertPool()
+		if !caCertPool.AppendCertsFromPEM(caCert) {
+			return nil, fmt.Errorf("failed to parse CA certificate from %s", promConfig.CACertPath)
+		}
+		config.RootCAs = caCertPool
+		logger.Log.Info("CA certificate loaded successfully", "path", promConfig.CACertPath)
+	}
+
+	// Load client certificate and key if provided
+	if promConfig.ClientCertPath != "" && promConfig.ClientKeyPath != "" {
+		cert, err := tls.LoadX509KeyPair(promConfig.ClientCertPath, promConfig.ClientKeyPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load client certificate from %s and key from %s: %w",
+				promConfig.ClientCertPath, promConfig.ClientKeyPath, err)
+		}
+		config.Certificates = []tls.Certificate{cert}
+		logger.Log.Info("Client certificate loaded successfully",
+			"cert_path", promConfig.ClientCertPath, "key_path", promConfig.ClientKeyPath)
+	}
+
+	return config, nil
+}
+
+// ValidateTLSConfig validates TLS configuration
+func ValidateTLSConfig(promConfig *interfaces.PrometheusConfig) error {
+	if promConfig == nil || !promConfig.EnableTLS {
```

Comment:
nil check here?

Reply:
I dont think we should send nil objects inside the function, a nil check should be a level higher

Reply:
removed